packages/formstr-app/src/containers/CreateForm/components/FormSettings/RelayList.tsx

```diff
@@ -0,0 +1,206 @@
+            </div>
+            <div>
+              <EditOutlined
+                onClick={() => {
+                  setEdit(true);
+                }}
+                className="relay-icons"
+              />
+              <DeleteOutlined className="relay-icons" onClick={handleDelete} />
+            </div>
+          </>
+        )}
+        {edit && (
+          <>
+            <CloseOutlined
+              onClick={() => {
+                setEdit(false);
+              }}
+              className="relay-edit-item"
+            />
+            <Input
+              defaultValue={relay.url}
+              onChange={handleInputChange}
+              className="relay-edit-item"
+            />
+            <Button
+              size="middle"
+              icon={<SaveOutlined className="relay-icon" />}
+              type="dashed"
+              onClick={onEditRelay}
+            />
+          </>
+        )}
+      </div>
+      {isError && <Text type="danger"> Not a valid websocket url</Text>}
+    </>
+  );
+};
+
+const RelayListItems = () => {
+  let relays = getDefaultRelays().map((relay) => {
+    return { url: relay, tempId: makeTag(6) };
+  });
+  const { relayList, setRelayList } = useFormBuilderContext();
+  const [newRelay, setNewRelay] = useState(false);
+  const [newRelayUrl, setNewRelayUrl] = useState("");
+  const [isError, setIsError] = useState(false);
+
+  const onAdd = () => {
+    if (!isValidWebSocketUrl(newRelayUrl)) {
+      setIsError(true);
+      return;
+    }
+    setIsError(false);
+    setRelayList([...relayList, { url: newRelayUrl, tempId: makeTag(6) }]);
+    setNewRelay(false);
+  };
+  const onEdit = (newRelay: string, tempId: string) => {
+    relays = relayList.map((relay) => {
+      if (relay.tempId === tempId) {
+        return { url: newRelay, tempId: tempId };
+      }
+      return relay;
+    });
+    setRelayList(relays);
+  };
+
+  const onDelete = (tempId: string) => {
+    relays = relayList.filter((relay) => relay.tempId !== tempId);
+    setRelayList(relays);
+  };
+
+  return (
+    <>
+      {newRelay && (
+        <div className="relay-item">
+          <CloseOutlined onClick={() => setNewRelay(false)} />
+          <Input
+            placeholder="Enter relay url"
+            onChange={(e) => {
+              setNewRelayUrl(e.target.value);
+            }}
+          />
+          <Button
+            size="middle"
+            icon={<SaveOutlined className="relay-icon" />}
+            type="dashed"
+            onClick={onAdd}
+          />
+        </div>
+      )}
+      {isError && <Text type="danger"> Not a valid websocket url</Text>}
+      <div className="add-relay-container">
+        {!newRelay && (
+          <Button
+            type="link"
+            onClick={() => {
+              setNewRelay(true);
+            }}
+          >
+            + Add Relay
+          </Button>
+        )}
+      </div>
+      {relayList.map((relay) => (
```

Comment:
we should save user relay list to the local storage or nostr profile?

Reply:
this is relay settings for each form, we should work on a users preferred relay  soon